### ISSUE TITLE
rpi-4.18.y ACT LED broken

### ISSUE THREAD

**Issue by asavah** [raspberrypi/linux#2704]
ACT LED does not seem to work at all on 4.18.y on RPI 3b+

Self built kernel, bcm2709_defconfig @ 3a1a31d
```
vcgencmd version
Sep 21 2018 15:48:28
Copyright (c) 2012 Broadcom
version 07f57128b8491ffdefcdfd13f7b4961b3006d9a9 (clean) (release)
```

The led itself works, when powering the pi or shutting it down it blinks as supposed to.
The sdcard activity however does not produce blinks.

`ls /sys/class/leds/` produces empty output.

lsgpio (built from the same tree/branch as the kernel)
```
lsgpio
GPIO chip: gpiochip0, "pinctrl-bcm2835", 54 GPIO lines
        line  0: unnamed unused
        line  1: unnamed unused
        line  2: unnamed unused
        line  3: unnamed unused
        line  4: unnamed unused
        line  5: unnamed unused
        line  6: unnamed unused
        line  7: unnamed unused
        line  8: unnamed unused
        line  9: unnamed unused
        line 10: unnamed unused
        line 11: unnamed unused
        line 12: unnamed "sysfs" [kernel]
        line 13: unnamed unused
        line 14: unnamed unused
        line 15: unnamed unused
        line 16: unnamed "sysfs" [kernel]
        line 17: unnamed unused
        line 18: unnamed "ir-receiver@12" [kernel active-low]
        line 19: unnamed "sysfs" [kernel]
        line 20: unnamed "sysfs" [kernel]
        line 21: unnamed "sysfs" [kernel]
        line 22: unnamed unused
        line 23: unnamed unused
        line 24: unnamed unused
        line 25: unnamed unused
        line 26: unnamed unused
        line 27: unnamed unused
        line 28: unnamed unused
        line 29: unnamed unused [output]
        line 30: unnamed unused
        line 31: unnamed unused
        line 32: unnamed unused
        line 33: unnamed unused
        line 34: unnamed unused
        line 35: unnamed unused
        line 36: unnamed unused
        line 37: unnamed unused
        line 38: unnamed unused
        line 39: unnamed unused
        line 40: unnamed unused
        line 41: unnamed unused
        line 42: unnamed unused
        line 43: unnamed unused
        line 44: unnamed unused
        line 45: unnamed unused
        line 46: unnamed unused
        line 47: unnamed unused
        line 48: unnamed unused
        line 49: unnamed unused
        line 50: unnamed unused
        line 51: unnamed unused
        line 52: unnamed unused
        line 53: unnamed unused
```

config.txt
```
start_x=1
disable_overscan=1
disable_splash=1
core_freq=500
gpu_freq=500
sdram_freq=580
sdram_schmoo=0x02000020
over_voltage_sdram=5
over_voltage=2
gpu_mem=320
dtoverlay=i2c-rtc,ds3231
dtoverlay=gpio-ir
dtparam=watchdog=on
dtparam=random=on
dtparam=audio=on
dtparam=spi=on
dtparam=i2c=on
hdmi_force_hotplug=1
decode_MPG2=snip
decode_WVC1=snip
hdmi_edid_file=1
dtparam=eee=off

```

**Comment by pelwell** [comment]
Thanks for reporting this. In rpi-4.18.y we're using the new, upstreamed GPIO expander driver. This has a different "compatible" string, and the .dts files hadn't been updated to match.

A fix has been pushed to rpi-4.18.y.

**Comment by asavah** [comment]
TYVM, works as expected.
